xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml

```diff
@@ -938,7 +942,8 @@
   #set ($filterQuery = "#livetable_getFilterQuery(""${safe_tableAlias}.value"" $matchType false $valueCount $paramPrefix)")
   #set ($whereSql = "${whereSql} and ($filterQuery.trim())")
   #foreach ($filterValue in $filterValues)
-    #livetable_addFilterParam($filterValue $matchType $whereParams "${paramPrefix}${foreach.count}")
+    #set ($actualMatchType = $request.getParameterValues("${colname}_match").get($foreach.count - 1))
```

Comment:
Do you need this? Doesn't ``$matchType`` already have the right value?

Reply:
Not when several filter are applied, with the previous code if you apply several matchers, only the first one is applied on all iterations.